tests/e2e/AdapterBase.php

```diff
@@ -58,4 +58,24 @@ public function testAdapter(): void
         $response = $logger->addLog($this->log);
         $this->assertEquals($this->expected, $response);
     }
+
+    /**
+     * @throws \Throwable
+     */
+    public function testSampler(): void
+    {
+        if (empty($this->log) || empty($this->adapter)) {
+            throw new \Exception('Log or adapter not set');
+        }
+        $logger = new Logger($this->adapter);
+        $logger->sample(0.1);
+
+        $results = [];
+
+        for ($x = 0; $x <= 10; $x++) {
+            $results[] = $logger->addLog($this->log);
+        }
+
+        $this->assertTrue(in_array(0, $results));
```

Comment:
This test is too flaky, there could be chances where random will not 100% work in 1 out of 10.